File: benchmarks/r/06_drug_utilisation.R
# Benchmark 06: Drug Utilisation
# R equivalent: DrugUtilisation::generateIngredientCohortSet() + summariseDrugUtilisation()

source("benchmarks/r/00_helpers.R")
library(DrugUtilisation)
cat("=== 06: Drug Utilisation ===\n")
t0 <- proc.time()

con <- dbConnect(duckdb(), dbdir = DB_PATH)
cdm <- cdmFromCon(con = con, cdmSchema = "main", writeSchema = "main", cdmName = "synthea_1k")

# Generate ingredient cohort for clopidogrel (1322184) — most common drug era
cdm <- generateIngredientCohortSet(
  cdm = cdm,
  name = "drug_cohort",
  ingredient = "clopidogrel"
)

result <- summariseDrugUtilisation(cdm$drug_cohort, ingredientConceptId = 1322184)
save_result(as.data.frame(result), "06_drug_utilisation")
elapsed <- (proc.time() - t0)[["elapsed"]]
save_timing("06_drug_utilisation", elapsed)
cdmDisconnect(cdm)
cat("Done in", elapsed, "seconds\n")

File: benchmarks/python/06_drug_utilisation.py
"""Benchmark 06: Drug Utilisation — omopy.drug"""

from helpers import Timer, connect_cdm, save_result, save_timing

from omopy.drug import generate_ingredient_cohort_set, summarise_drug_utilisation

print("=== 06: Drug Utilisation ===")
t = Timer()
cdm = connect_cdm()

cdm = generate_ingredient_cohort_set(cdm, name="drug_cohort", ingredient="clopidogrel")
cohort = cdm.cohort_tables["drug_cohort"]

result = summarise_drug_utilisation(cohort, ingredient_concept_id=1322184, gap_era=30)
save_result(result.data, "06_drug_utilisation")
save_timing("06_drug_utilisation", t.elapsed())
print(f"Done in {t.elapsed():.2f} seconds")
